sources/packages/backend/apps/api/src/route-controllers/application/application.students.controller.ts

```diff
@@ -664,4 +665,35 @@ export class ApplicationStudentsController extends BaseController {
       },
     );
   }
+
+  /**
+   * Get application overall details for the given application.
+   * @param applicationId application Id.
+   * @returns application overall details.
+   */
+  @ApiNotFoundResponse({
+    description: "Application not found.",
+  })
+  @Get(":applicationId/overall-details")
+  async getApplicationOverallDetails(
+    @Param("applicationId", ParseIntPipe) applicationId: number,
+    @UserToken() userToken: StudentUserToken,
+  ): Promise<ApplicationOverallDetailsAPIOutDTO> {
+    const application = await this.applicationService.doesApplicationExist({
+      applicationId: applicationId,
+      studentId: userToken.studentId,
+    });
+    if (!application) {
+      throw new NotFoundException("Application not found.");
+    }
+    const applications =
+      await this.applicationService.getAllApplicationVersions(applicationId);
+    return {
+      previousVersions: applications.map((application) => ({
+        id: application.id,
+        submittedDate: application.submittedDate,
+        applicationEditStatus: application.applicationEditStatus,
+      })),
+    };
+  }
```

Comment:
What is the reason for not centralizing the code with controller service?

Reply:
No reason; I can add it to the controller service.
I will also refactor to prevent the two DB calls by removing the `doesApplicationExist`.

Reply:
The code was refactored to avoid the two DB queries and moved to a centralized place.
If the methods evolve differently for students and the Ministry, we can review the centralization as needed.